src/common/util.js

```diff
@@ -185,10 +251,22 @@ function isNextLineEmptyAfterIndex(text, index) {
   return hasNewline(text, idx);
 }
 
+/**
+ * @template N
+ * @param {string} text
+ * @param {N} node
+ * @param {(node: N) => number} locEnd
+ * @returns {boolean}
+ */
 function isNextLineEmpty(text, node, locEnd) {
   return isNextLineEmptyAfterIndex(text, locEnd(node));
 }
 
+/**
+ * @param {string} text
+ * @param {number} idx
+ * @returns {number}
```

Comment:
Should this be `@returns {number}` or `@returns {number | false}`?

Reply:
This function returns `idx`. `idx` contains the result of `skip*` functions. Since they return `number | false`, so must this function, unless I’m missing something. So `@returns {number | false}`